### DISCUSSION TITLE
Getting wrong HTTP status

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#2704] [Q&A] [answered]
Hello, one of my collections has a update rule like this: 
```
id = @request.auth.id
```

So, if a non authorized user tries to update a record, they should get `403 Forbidden`. However, pocketbase is responding with `404 Not Found`. Is this the expected behavior? If it is, are there any work-around to show user the appropriate status?

**Comment by ganigeorgiev** [comment]
It is working as intended because the collection API rules act also as filters (eg. you can have `@request.auth.id = id || status = "public"`) and there is no easy and efficient way at the moment to detect on which exact part of the rule the query has failed because the rule is executed in one pass.

> If it is, are there any work-around to show user the appropriate status?

You can overwrite the response manually with a `OnRecordBefore*` hook, but I personally wouldn't botter with that. Why the 403 status code is important for your use case? Someone could argue also that from security standpoint returning 404 is maybe more safe as it prevents accidental information disclosure that the resource exists.

**Reply by Mr-M1M3** [reply]
> Why the 403 status code is important for your use case?

When a user tries to change the password with incorrect old password, shouldn't we tell the user that the password is wrong rather than telling that the resource wasn't found?

**Reply by ganigeorgiev** [reply, marked as answer]
If the user have access to the resource and it is existing, you should get a validation error with the failed submitted field(s).

**Reply by Mr-M1M3** [reply]
I get it now. But I definitely wanted to update password and it showed 404. Not sure why it's working now as you said. I think I'm going crazy 🙂

**Reply by ganigeorgiev** [reply]
It is possible to get 404 if the request was submitted without being authenticated or the token was expired.

**Reply by Mr-M1M3** [reply]
Ahh... Now I get it. But, I'm adding old password in the request body. So, the token shouldn't matter, right?

**Reply by ganigeorgiev** [reply]
The `oldPassword` is a regular request body field. Providing it doesn't authenticate the user. A request is considered authenticated only if it was sent with the `Authorization: TOKEN` header.
